ibc-testkit/src/testapp/ibc/core/types.rs

```diff
@@ -164,22 +152,20 @@ where
 
 /// A context implementing the dependencies necessary for testing any IBC module.
 #[derive(Debug)]
-pub struct MockGenericContext<S>
+pub struct MockGenericContext<S, H>
 where
     S: ProvableStore + Debug,
+    H: TestHost,
 {
     /// The type of host chain underlying this mock context.
-    pub host_chain_type: HostType,
-
-    /// Host chain identifier.
-    pub host_chain_id: ChainId,
+    pub host: H,
 
     /// Maximum size for the history of the host chain. Any block older than this is pruned.
     pub max_history_size: u64,
 
     /// The chain of blocks underlying this context. A vector of size up to `max_history_size`
     /// blocks, ascending order by their height (latest block is on the last position).
-    pub history: Vec<HostBlock>,
+    pub history: Vec<H::Block>,
 
     /// Average time duration between blocks
     pub block_time: Duration,
```

Comment:
Should we include these fields (`max_history_size`, `history`, `block_time`) under the `host` in this PR, or is it planned to address this in a subsequent update?

Reply:
Great idea! You already suggested above. Let's tackle this in the next PR. In this PR, I focused on generalizing the `HostType`/`HostBlock` enum using traits.